### ISSUE TITLE
Timeout param not passed to requests library

### ISSUE THREAD

**Issue by RedBlueThing** [NeverBounce/NeverBounceApi-Python#11]
* NeverBounceApi-Python version: 4.2.6
* Python version: 3.6.1
* Operating System: MacOS

### Description

I pass a timeout to the library to ensure neverbounce requests do not block our system for longer than that timeout. The timeout I pass to the single_check method is not being passed to the requests library so when the Neverbounce infrastructure is unresponsive requests do not timeout correctly

### What I Did

```
single_check("foo@bar.com", timeout=1)
```

I used a proxy to simulate unresponsive infrastructure. When I make the call I get a response rather than the request timing out after 1 second.

This bug in the library can be fixed with the following change to line 37 of single.py:

```
resp = self._make_request('GET', endpoint, params=params, timeout=timeout)
```

(It is passing the timeout through in the params dictionary, but that is just forwarded to the Neverbounce servers and not passed to the requests library). So the change is just to add the timeout param to the _make_request arguments (which are passed on to requests.request via kwargs).

**Comment by mmollick** [comment]
Hello @RedBlueThing,

The single endpoint dopes accept a [timeout parameter](https://developers.neverbounce.com/v4.0/reference#single-check) in the request. This timeout parameter indicates the maximum amount of time we'll spend verifying an email before giving up and assuming it's a `unknown`. 

With that said, I understand why you would want to specify a request timeout on the client. I'm introducing a new `timeout` parameter on the client, it can either be specified along with the `api_key` when instantiating it or by setting the parameter directly. Here's the PR for this update, let me know if this will work: https://github.com/NeverBounce/NeverBounceApi-Python/pull/12

**Comment by RedBlueThing** [comment]
Thanks Mike. Looks good to me.

**Comment by mmollick** [comment]
I've released #12 as version [4.2.7](https://pypi.org/project/neverbounce-sdk/4.2.7/)

**Comment by RedBlueThing** [comment]
Thanks Mike!
